src/Microsoft.AspNetCore.Razor.Language/RazorConfiguration.cs

```diff
@@ -44,18 +44,38 @@ public sealed class RazorConfiguration
                 throw new ArgumentNullException(nameof(extensions));
             }
 
-            LanguageVersion = languageVersion;
-            ConfigurationName = configurationName;
-            Extensions = extensions.ToArray();
-            DesignTime = designTime;
+            return new DefaultRazorConfiguration(languageVersion, configurationName, extensions.ToArray(), designTime);
         }
 
-        public string ConfigurationName { get; }
+        public abstract string ConfigurationName { get; }
 
-        public IReadOnlyList<RazorExtension> Extensions { get; }
+        public abstract IReadOnlyList<RazorExtension> Extensions { get; }
 
-        public RazorLanguageVersion LanguageVersion { get; }
+        public abstract RazorLanguageVersion LanguageVersion { get; }
 
-        public bool DesignTime { get; }
+        public abstract bool DesignTime { get; }
+
+        private class DefaultRazorConfiguration : RazorConfiguration
+        {
+            public DefaultRazorConfiguration(
+                RazorLanguageVersion languageVersion,
+                string configurationName,
+                RazorExtension[] extensions,
+                bool designTime)
+            {
+                LanguageVersion = languageVersion;
+                ConfigurationName = configurationName;
+                Extensions = extensions;
+                DesignTime = designTime;
+            }
+
+            public override string ConfigurationName { get; }
+
+            public override IReadOnlyList<RazorExtension> Extensions { get; }
```

Comment:
Interesting. When you first added RazorExtension I didn't see that it was in the Razor.Language assembly. Gots me curious 😄